app/components/calculate-time.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { Ruleset } from "../interfaces/Ruleset";
+import { EnterHours } from "../interfaces/EnterHours";
+import { calculateHours, getTimeValue } from '../utils/utils';
+
+interface CalculateTimeProps {
+  rulesets: Ruleset[];
+  hoursWorked: EnterHours[];
+}
+
+export function calculateTotalTOIL(rulesets: Ruleset[], hoursWorked: EnterHours[]): number {
+  let totalTOIL = 0;
+  hoursWorked.forEach((hours) => {
+    const hoursDayOfWeek = hours.date.toLocaleString('en-US', { weekday: 'long' });
+    const matchingRuleset = rulesets.find((ruleset) =>
+      // Compare ruleset dayOfWeek and startTime with hours date and startTime
+      ruleset.dayOfWeek === hoursDayOfWeek &&
+      getTimeValue(ruleset.startTime) <= getTimeValue(hours.startTime) &&
+      getTimeValue(ruleset.endTime) >= getTimeValue(hours.endTime)
+    );
+
+    if (matchingRuleset) {
+      const hoursWorked = calculateHours(hours.startTime, hours.endTime);
+      totalTOIL += hoursWorked * matchingRuleset.multiplier;
```

Comment:
**thought:** I know this is non blocking but I can see there is a todo to highlight the maths - we could capture this calc somehow and display this in that box --> as we already know that this is the matching ruleset. May be a bit finiky but could return and maybe a separate function is more appropriate.